### ISSUE TITLE
Logs Visualizations: Download logs respect column filters

### ISSUE THREAD

**Issue by gtk-grafana** [grafana/grafana#77817]
Currently in the logs table UI any customizations made in the table are not reflected in the downloaded logs (txt or json)

- [x] Logs Panel support
- [ ] Logs Table support

**Comment by gtk-grafana** [comment]
First thoughts after poking around in the code: Download UI is under LogsMetaRow.tsx which is a child of Logs.tsx, `downloadLogsModelAsTxt` in download.ts and `logRowsToReadableJson`in utils.ts are doing the work of converting the `logsRowModel` to a txt/json file.
These functions can also be called in the query inspector.

Might be trouble dealing with `logsRowModel` when we want to transform, the existing components don't seem to have the original dataframe. 
I need to look into using the selected columns to filter the logsRowModel, if it's extracting the labels already there might be a quick solution available

**Comment by gtk-grafana** [comment]
Yeah we just spit out the raw entry and a timestamp for each row. We'd need to pull up the state of the modified dataFrame from the LogsTableWrap to Logs, and then convert the download functions to take dataFrames instead of logFrames.

**Comment by neillfontes-klar** [comment]
Hi,

I wanted to extract a list of `userId`s from a query result and feed it into a script for doctoring these users in my system. However the CSV output is polluted with several other fields.

This is my query:

```
{container="redacted", instance="redacted", level="ERROR"} 
| keep message
|= "redacted"
| regexp "user (?P<userId>[a-f0-9-]+)"
| line_format "{{.userId}}"
```

But the output CSV comes as:

```
"Time","Line","tsNs","id","TraceID","userId"
1736350429897,ffbe23a3-d1b3-4929-8a27-ef782e308a1e,1736350429897806333,1736350429897806333_8d263c9a,,ffbe23a3-d1b3-4929-8a27-ef782e308a1e
1736350203297,ffb2c49c-fdd6-4e40-9965-651fe2ca5047,1736350203297867562,1736350203297867562_4401fb76,,ffb2c49c-fdd6-4e40-9965-651fe2ca5047
1736350191251,fd478fa7-a48f-438d-9ceb-52b8e6c85ffe,1736350191251958653,1736350191251958653_2529087e,,fd478fa7-a48f-438d-9ceb-52b8e6c85ffe
```

I couldn't find a way to export just the `userId`s into a single column alone. That would be a massive help when dealing with system issues and so on. This way I don't need to fiddle with `awk` scripts and so on. 

Is there an option to do so?

Thanks in advance.

**Comment by gtk-grafana** [comment]
@neillfontes-klar unfortunately there isn't a way to do this in the UI at this time, however [logCLI](https://grafana.com/docs/loki/latest/query/logcli/getting-started/) might provide you with a bit more flexibility over the response

**Comment by neillfontes-klar** [comment]
Alright, I leave this as a UX suggestion then as it is a pretty standard use case for several engineers I bet?

**Comment by matyax** [comment]
Considering this done in the Table NG.
